File: packages/posthog/android/src/main/java/io/capawesome/capacitorjs/plugins/posthog/classes/options/ScreenOptions.java
package io.capawesome.capacitorjs.plugins.posthog.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.posthog.PosthogHelper;
import java.util.Map;
import org.json.JSONException;

public class ScreenOptions {

    @NonNull
    private String screenTitle;

    @Nullable
    private Map<String, Object> properties;

    public ScreenOptions(@NonNull String screenTitle, @Nullable JSObject properties) throws JSONException {
        this.screenTitle = screenTitle;
        this.properties = PosthogHelper.createHashMapFromJSONObject(properties);
    }

    @NonNull
    public String getScreenTitle() {
        return screenTitle;
    }

    @Nullable
    public Map<String, Object> getProperties() {
        return properties;
    }
}

File: packages/posthog/ios/Plugin/Classes/Options/ScreenOptions.swift
import Foundation
import Capacitor

@objc public class ScreenOptions: NSObject {
    private var screenTitle: String
    private var properties: [String: Any]?

    init(screenTitle: String, properties: JSObject?) {
        self.screenTitle = screenTitle
        self.properties = PosthogHelper.createHashMapFromJSObject(properties)
    }

    func getScreenTitle() -> String {
        return screenTitle
    }

    func getProperties() -> [String: Any]? {
        return properties
    }
}
